osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs

```diff
@@ -136,6 +137,10 @@ public Drawable GetDrawableComponent(ISkinComponent component)
                                 return SkinUtils.As<TValue>(new BindableFloat(LEGACY_CIRCLE_RADIUS));
 
                             break;
+
+                        case OsuSkinConfiguration.ExpandNumberPiece:
+                            bool expand = !(source.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value >= 2.0m);
```

Comment:
are you sure this is correct? won't it stop the propagation of skin lookup at the first skin?

Reply:
in addition, this *definitely* needs an in-line comment explaining this. ie. the fact that skin version 2.0 changed behaviour.

else it actually reads like it's a bug (and there's a missing configuration lookup).

Reply:
> are you sure this is correct? won't it stop the propagation of skin lookup at the first skin?

I don't think there's an issue with that, the `Version` is the lookup that would be propagated until top-parent, same is done with:
https://github.com/ppy/osu/blob/ccc20a93412ead98c9e80f2cdcd4b911ff703935/osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs#L93-L104

Reply:
While we're on this I find reading `!(source.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value >= 2.0m);` super backwards. Couldn't this just be a `<` so that there's no need to mentally flip this check when reading this code?

Reply:
@bdach because C# has a very weird thing with nullables, check out the full convo. of https://github.com/ppy/osu/pull/6345#discussion_r358596459, I'm gonna try my best to split this.

Reply:
I don't recall reading that, fair enough. I guess it can stay then to not belabour this further. Maybe leave an inline comment or something explaining why this is written this weirdly.